tests/phpunit/PaymentMethods/WC_Stripe_Express_Checkout_Element_Test.php

```diff
@@ -431,4 +426,84 @@ public function test_add_order_attribution_data() {
 		$output = ob_get_clean();
 		$this->assertStringMatchesFormat( '%aid="wc-stripe-express-checkout__order-attribution-inputs"%a', $output );
 	}
+
+	/**
+	 * Test for `is_product_page_for_ece`.
+	 *
+	 * The is_product_page_for_ece method determines whether ECE should use product
+	 * pricing vs cart pricing. It returns false when:
+	 * - Not on a product page
+	 * - On a One Page Checkout page with ECE enabled on checkout (should use cart pricing)
+	 *
+	 * @param bool $is_product               Whether we're on a product page.
+	 * @param bool $is_opc                   Whether One Page Checkout is active.
+	 * @param bool $show_ece_on_checkout     Whether ECE is enabled on checkout pages.
+	 * @param bool $expected_result          Expected return value from is_product_page_for_ece.
+	 * @return void
+	 * @dataProvider provide_test_is_product_page_for_ece
+	 */
+	public function test_is_product_page_for_ece( $is_product, $is_opc, $show_ece_on_checkout, $expected_result ) {
+		$ajax_handler = $this->getMockBuilder( WC_Stripe_Express_Checkout_Ajax_Handler::class )
+			->disableOriginalConstructor()
+			->getMock();
+
+		$helper = $this->getMockBuilder( WC_Stripe_Express_Checkout_Helper::class )
+			->disableOriginalConstructor()
+			->setMethods( [ 'is_product', 'is_one_page_checkout', 'should_show_ece_on_checkout_page' ] )
```

Comment:
Nit: `setMethods()` was removed in PHPUnit 10, so it will be better to use `onlyMethods()` to avoid future work.
```suggestion
			->onlyMethods( [ 'is_product', 'is_one_page_checkout', 'should_show_ece_on_checkout_page' ] )
```